tg_bot.py

```diff
@@ -40,8 +41,23 @@ def display_tweet(self, context):
         job = context.job
         global KILL_SIGNAL
         tid = pd.read_csv(self.dirpath + 'tid_store.csv')
-        for index, tid_store in tid.iterrows():
-
+        tid["Date_time"]= pd.to_datetime(tid["Date_time"])
+        previous_date = tid['Date_time'].iloc[-1]
+        
+        #to find the previous date in the tid
+        for index, tid_store in tid[::-1].iterrows():
+            scrape_datetime = tid_store['Date_time']
+            scrape_date = date(scrape_datetime.year, scrape_datetime.month, scrape_datetime.day)
+            if scrape_date!=previous_date:
+                previous_date = scrape_date
+                break
+            else:
+                continue
+        
+        for index, tid_store in tid[::-1].iterrows():
+            scrape_date = tid_store['Date_time']
+            if scrape_date<=previous_date:
```

Comment:
Why is there two loops and checks for basically the same purpose? Isn't it redundant? doesn't adding the condition in the above loop works well?

Reply:
Well we need to find the previous date before comparing in the second loop. Here the previous date is the date just before the latest date, which is the limit till which tweets will be sent. If we add it to the second loop it won't work.

Reply:
What I'm saying is, the below two lines are similar. So the logic tends to be similar. Please look into the code a little bit more to find similarities and remove redundant lines of code. Since there is two loops which do the exact same thing, I believe that would increase the complexity as well.

```
        for index, tid_store in tid[::-1].iterrows():
            scrape_date = tid_store['Date_time']
```

Reply:
If I understood you right, you mean to change the variable names and some similarities in the loops or just remove the first loop?